lib/resetState.ts

```diff
@@ -0,0 +1,61 @@
+type SerializableObjVal =
+  | SerializableObj
+  | string
+  | number
+  | boolean
+  | null
+  | SerializableObjVal[];
+
+type SerializableObj = {
+  [key in string]: SerializableObjVal;
+};
+
+let stateId = 0;
+function genStateId(): number {
+  stateId += 1;
+  return stateId;
+}
+
+const STATE_ID_PROPERTY = "___testStateStateId";
+
+let stateIdToState: Map<number, SerializableObj> = new Map();
+
+function deepCopy(v: SerializableObj): SerializableObj {
+  // TODO: Recursively clone deeply.
+  return JSON.parse(JSON.stringify(v)) as SerializableObj;
+}
+
+function objectStateInitialized(obj: SerializableObj): boolean {
+  return STATE_ID_PROPERTY in obj;
+}
+
+function resetToInitialState(obj: SerializableObj): void {
+  const stateId = (obj as any)[STATE_ID_PROPERTY];
+  const initState = deepCopy(stateIdToState.get(stateId) ?? {});
```

Comment:
Why `?? {}`? If there's no state with the given `stateId` then probably the error should be thrown as this is not an expected situation.

Reply:
you are right, I just wanted to satisfy the ts compiler after changing `deepCopy` signature without polishing this code; but now fixed - method will throw